libs/common-utils/src/environments.ts

```diff
@@ -3,9 +3,11 @@ import { registerOnWindow } from './misc'
 const DEFAULT_ENVIRONMENTS_REGEX: Record<EnvironmentName, string> = {
   local: '^(:?localhost:\\d{2,5}|(?:127|192)(?:\\.[0-9]{1,3}){3})',
   pr: '^((?:explorer|swap)-dev-git-[\\w\\d-]+|swap-\\w{9}-)cowswap-dev\\.vercel\\.app',
-  development: '^(dev.swap.cow.fi|swap-develop.vercel.app)',
-  staging: '^(staging.swap.cow.fi|swap-staging.vercel.app)',
-  production: '^(swap.cow.fi|swap-prod.vercel.app)$',
+  development:
+    '^(dev.swap.cow.fi|dev.explorer.cow.fi|swap-develop.vercel.app|explorer-dev-git-develop-cowswap-dev.vercel.app)',
+  staging:
+    '^(staging.swap.cow.fi|staging.explorer.cow.fi|swap-staging.vercel.app|explorer-dev-git-main-cowswap-dev.vercel.app)',
+  production: '^(swap.cow.fi|explorer.cow.fi|swap-prod.vercel.app)$',
   barn: '^(barn.cow.fi|swap-barn.vercel.app)$',
```

Comment:
For completion, here's barn

<img width="338" height="208" alt="image" src="https://github.com/user-attachments/assets/9b0ce298-9bed-4545-a3b1-664323faee80" />

<img width="598" height="288" alt="image" src="https://github.com/user-attachments/assets/501d32c2-8c44-4750-a583-4c028f7a52cb" />